### PR TITLE
Add Cortical.io integration

### PR DESCRIPTION
**Pull request by kuzin-cortical**
This a **review-only** PR. After it is approved we'll open same PR for the EdenAI repo

### CODE REVIEW

File: edenai_apis/apis/corticalio/client.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    """
+    REST client for the Cortical.io NLP API
+    """
+
+    def __init__(self, api_settings: Dict) -> None:
+        self.api_settings = api_settings
+        self.auth_headers = {"Authorization": self.api_settings.get("api_key")}
+        self.base_url = self.api_settings.get("base_url").rstrip('/')
+
+    def extract_keywords(self, text: str, language: str):
```
Comment (hhobel) [inline edenai_apis/apis/corticalio/client.py:20]: I don't know how actual clients are using this but documentation but we usually document public methods

File: edenai_apis/apis/corticalio/client.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            url=f"{self.base_url}/keywords",
+            headers=self.auth_headers,
+            json={
+                "text": json.dumps(text),
+                "language": language
+            }
+        )
+        return self.handle_response(response)
+
+    def handle_response(self, response: Response):
```
Comment (hhobel) [inline edenai_apis/apis/corticalio/client.py]: should be private?

File: edenai_apis/apis/corticalio/corticalio_api.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    provider_name = "corticalio"
+
+    def __init__(self, api_keys: Dict = None):
+        api_keys = api_keys or {}
+        self.api_settings = load_provider(
+            ProviderDataEnum.KEY, self.provider_name, api_keys=api_keys
+        )
+        self.client = CorticalClient(self.api_settings)
+
+    def text__keyword_extraction(
```
Comment (hhobel) [inline edenai_apis/apis/corticalio/corticalio_api.py:26]: should there really be two underscores? Documentation?

File: edenai_apis/apis/corticalio/helpers.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+def normalize_keywords(response: Dict):
+    items = [
+        InfosKeywordExtractionDataClass(
+            keyword=keyword["word"],
+            importance="{:.2f}".format(keyword["score"])
+        ) for keyword in response["keywords"]
+    ]
+
+    return KeywordExtractionDataClass(items=items)
```
Comment (hhobel) [inline edenai_apis/apis/corticalio/helpers.py:14]: I would actually move this into the corticalio_api.py but if it is the general coding schema it's also fine

**Review by hhobel — APPROVED** [review]
LGTM! I don't know how edenAI and their developers, users are handling this but I would at least document the endpoint extensively (telling exactly what the endpoint is doing). I noticed that language is not optional - we should at least document which error you get if the language is not supported and which format we need (ISO, etc).

**Review by tjagvad — APPROVED** [review]
lgtm

File: edenai_apis/apis/corticalio/corticalio_api.py (hunk 3 above)
Comment (kuzin-cortical) [reply edenai_apis/apis/corticalio/corticalio_api.py:26]: yes, it's implementation of an abstract method from the base class

File: edenai_apis/apis/corticalio/helpers.py (hunk 4 above)
Comment (kuzin-cortical) [reply edenai_apis/apis/corticalio/helpers.py:14]: `corticalio_api.py` is an single point of implementation of all supported features (via abstract base classes) and will potentially get quite huge

File: edenai_apis/apis/corticalio/client.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    """
+    REST client for the Cortical.io NLP API
+    """
+
+    def __init__(self, api_settings: Dict) -> None:
+        self.api_settings = api_settings
+        self.auth_headers = {"Authorization": self.api_settings.get("api_key")}
+        self.base_url = self.api_settings.get("base_url").rstrip('/')
+
+    def extract_keywords(self, text: str, language: str):
```
Comment (hhobel) [inline edenai_apis/apis/corticalio/client.py:20]:
```suggestion
    def extract_keywords(self, text: str, language: Optional[str]):
```
Are you sure that it works like this if language is optional?

Comment (HubertBaginski) [reply edenai_apis/apis/corticalio/client.py:20]: we autodetect language and/or should default to english irrc?

File: edenai_apis/apis/corticalio/client.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        """
+        Extract keywords from the provided text.
+
+        Args:
+            text (str): text of interest, required
+            language (str): language of the specified text, optional
```
Comment (HubertBaginski) [inline edenai_apis/apis/corticalio/client.py:26]: update docstring for language if optional

File: edenai_apis/apis/corticalio/corticalio_api.py [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+from edenai_apis.loaders.data_loader import ProviderDataEnum
+from edenai_apis.loaders.loaders import load_provider
+from edenai_apis.utils.types import ResponseType
+
+
+class CorticalioApi(
+    ProviderInterface,
+    TextInterface
+):
+    provider_name = "corticalio"
```
Comment (HubertBaginski) [inline edenai_apis/apis/corticalio/corticalio_api.py:17]: upper case constant? or was there a reason why we chose lower case
```suggestion
    PROVIDER_NAME = "corticalio"
```

**Review by HubertBaginski — APPROVED** [review]
LGTM!

File: edenai_apis/apis/corticalio/corticalio_api.py (hunk 7 above)
Comment (kuzin-cortical) [reply edenai_apis/apis/corticalio/corticalio_api.py:17]: Yes, it has to be lower case or some tests from Eden are going to fail

**Pull request closed without merging**